### PR TITLE
Tech docs

### PR DESCRIPTION
**Pull request by SaFrMo**
Here's a start to some tech documentation for the project! Hopefully this will make jumping in and working a little easier. Let me know if anything looks off-base or missing and I'll take a look - thanks!

### CODE REVIEW

File: tech-notes.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+## Tech Notes
+
+There are 6 canvases on the page:
+
+* `imageCanvas` is the only visible canvas. It's where the composite image is rendered and is downloaded when the user is done.
+* `tempCanvas` holds path data when drawing paths.
+* `holderCanvas` takes a snapshot of the `imageCanvas` as soon as a new path starts.
+* `rotationCanvas`
```
Comment (bMacSwigg) [inline tech-notes.md:8]: What's the purpose of rotationCanvas?

File: tech-notes.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+### Drawing a path
+
+When you click or tap on the canvas:
+
+1. **On mouse down/tap start:**
+    1. The current `imageCanvas` image is copied to the `holderCanvas`.
+    1. The `tempCanvas` is cleared. 
+    1. The mouse/touch position is saved as `lastPos`.
+1. **On mouse/touch move:**
+    1. A circle is drawn at each point to the `imageCanvas` and `tempCanvas`.
```
Comment (bMacSwigg) [inline tech-notes.md:30]: And a rectangle is interpolated between the new position and `lastPos`

File: tech-notes.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    1. The mouse/touch position is saved as `lastPos`.
+1. **On mouse/tap end (or the cursor leaving the canvas):**
+    1. *If the user is painting a solid color:*
+        * No extra action is taken, since the solid colors were drawn onto the `imagCanvas` during the mouse/tap move step.
```
Comment (bMacSwigg) [inline tech-notes.md]: Nit: typo, "imagCanvas"

File: README.md [hunk 4]
```diff
@@ -43,3 +43,7 @@ Thank you:
 * And to everyone who has contributed to this repository or filed a bug report.
 
 My code is still a bit of a mess but utility over cleanliness, will continue to fix up in the coming days after the more critical work is over.
+
+## Tech notes
+
+See [this doc](/blob/master/tech-notes.md) for tech notes.
```
Comment (bMacSwigg) [inline README.md:49]: What are the expectations for keeping the tech notes up-to-date? e.g., should each pull request be required to also include any relevant changes to tech-notes.md?

File: tech-notes.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    1. *If the user is painting a blur:*
+        * The image from the `rotationCanvas` is copied to the `blurredCanvas`.
+        * The `blurredCanvas` image is pixelated (`pixelateCanvas`), then blurred (`stackBlurCanvasRGBA`).
+        * The image from the `blurredCanvas` is drawn onto the `tempCanvas`, using the alpha from the existing `tempCanvas` image as a mask ([source-in](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation) composite operation). This creates a blurred image that matches the path drawn on mouse/touch move.
+        * The image from the `tempCanvas` is drawn onto the `imageCanvas`.
+        * The image from the `holderCanvas` is drawn onto the `imagCanvas` below the existing content.
```
Comment (bMacSwigg) [inline tech-notes.md]: Nit: typo, "imagCanvas"

File: tech-notes.md (hunk 2 above)
Comment (SaFrMo) [reply tech-notes.md:30]: Noted and added, thanks!

File: README.md (hunk 4 above)
Comment (SaFrMo) [reply README.md:49]: That's a really good idea - @everestpipkin could that be a condition of accepting a PR?

File: tech-notes.md (hunk 1 above)
Comment (SaFrMo) [reply tech-notes.md:8]: I was wondering that too - it looks like it's never actually used, per [this note](https://github.com/SaFrMo/image-scrubber/blob/feature/docs-v2/tech-notes.md#rotating-an-image). I can definitely look into removing unused code like this over the next few days!

File: README.md (hunk 4 above)
Comment (everestpipkin) [reply README.md:49]: Yes, I'd appreciate it if folks will document their own changes! We can request it in the tech notes maybe. :) That said, I can also try to add notes as things are changes/updated if I see they're missing.

File: tech-notes.md (hunk 1 above)
Comment (everestpipkin) [reply tech-notes.md:8]: Oh, its actually used for a few things- both the main blur draw function and to save the canvas-data for the rotation function. My guess is we could take this project's canvas usage down by at least one offscreen canvas, but we'd need to map out the way the image flows between all of them for all the different saves, etc.. it got a bit twisted after a few layers of features and PRs.

Comment (everestpipkin) [reply tech-notes.md:8]: Oops, sorry, went to edit the readme and did it in your fork @SaFrMo !

Comment (SaFrMo) [reply tech-notes.md:8]: Haha no worries!

**Pull request merged**